apps/website-25/src/pages/api/courses/exercises/[exerciseId].ts

```diff
@@ -18,10 +18,10 @@ export default makeApiRoute({
     exercise: z.any(),
   }),
 }, async (body, { raw }) => {
-  const { courseSlug, unitId, exerciseId } = raw.req.query;
+  const { exerciseId } = raw.req.query;
 
   const exercise = (await db.scan(exerciseTable, {
-    filterByFormula: `AND({[>] Course slug} = "${courseSlug}", {[>] Unit number} = ${unitId}, {[*] Record ID} = "${exerciseId}")`,
+    filterByFormula: `AND({[*] Record ID} = "${exerciseId}")`,
   }))[0];
 
```

Comment:
The db.scan query could return an empty array if the exercise does not exist, and then getting `[0]` will result in `undefined` being sent back to the client.

You might want to add a check like:

```
if (!exercise) {
  throw createHttpError.NotFound("Exercise not found")
}
```